collection/collection/src/macMain/kotlin/androidx.collection.internal/Lock.mac.kt

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright 2022 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.collection.internal
```

Comment:
yea honestly i wasn't sure about what to name these since the default sourcesets use the mac prefix, hence i went with mac. happy to change if you feel strong about it.

Reply:
`macXyx`  is only for macOs, right? I suppose there will be `iosXyz`, `tvosXyz`, etc. So `macMain` sounds like being common for `macXyz` targets only.

Reply:
since we don't have ios targets yet, i didn't add for them (had a similar thought between androidNative vs linuxNative).
but i agree that this is for darwin hence should be called darwin, i'll make the change.

Reply:
done.